x/ibc/23-commitment/value.go

```diff
@@ -0,0 +1,89 @@
+package commitment
+
+import (
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/store/state"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+type Base struct {
+	cdc    *codec.Codec
+	prefix []byte
+}
+
+func NewBase(cdc *codec.Codec) Base {
+	return Base{
+		cdc: cdc,
+	}
+}
+
+func (base Base) Store(ctx sdk.Context) Store {
+	return NewPrefix(GetStore(ctx), base.prefix)
+}
+
+func join(a, b []byte) (res []byte) {
+	res = make([]byte, len(a)+len(b))
+	copy(res, a)
+	copy(res[len(a):], b)
+	return
+}
+
+func (base Base) Prefix(prefix []byte) Base {
+	return Base{
+		cdc:    base.cdc,
+		prefix: join(base.prefix, prefix),
+	}
+}
+
+type Mapping struct {
+	base Base
+}
+
+func NewMapping(base Base, prefix []byte) Mapping {
+	return Mapping{
+		base: base.Prefix(prefix),
+	}
+}
+
+type Value struct {
```

Comment:
this name is very general. Can you add a comment to what is this for?

Reply:
I don't really understand this abstraction either.